exploration/bidi-usability.md

```diff
@@ -348,69 +412,112 @@ the results or debug what is wrong with their messages.
+               / LRI "@" identifier [bidi] [[s] "=" [s] ((literal / variable) [bidi])] close-isolate
+markup         = "{" [s] "#" identifier [bidi] *(s option) *(s attribute) [s] ["/"] "}"  ; open and standalone
+               / "{" LRI [s] "#" identifier [bidi] *(s option) *(s attribute) [s] ["/"] close-isolate "}"
+               / "{" [s] "/" identifier [bidi] *(s option) *(s attribute) [s] "}"  ; close
+               / "{" LRI [s] "/" identifier [bidi] *(s option) *(s attribute) [s] close-isolate "}"  ; close
+identifier     = [(namespace ns-separator)] name
+ns-separator   = [bidi] ":"
+bidi           = [ %x200E-200F / %x061C ]
 ```
 
+
+### Isolate `name` rather than `unquoted-literal`
+
 Isolating rather than marking `name` helps ensure
 that its directionality does not spill over to adjoining syntax.
+
+The following replaces the proposed design's changes to `literal` and the `[bidi]` additions to
+`variable-expression`, `function`, `option`, `attribute`, `markup`, and `ns-separator`:
+```abnf
+name = [open-isolate] name-start *name-char [close-isolate]
+quoted-literal = [open-isolate] "|" *(quoted-char / quoted-escape) "|" [close-isolate]
+```
+
 For example, this allows for the proper rendering of the expression
 ```
 {⁦:⁧אחת⁩:⁧שתיים⁩⁩}
 ```
 where "אחת" is the `namespace` of the `identifier`.
-Without `name` isolation, this would render as
+Without `name` isolation, this would (misleadingly) render as
 ```
 {⁦:אחת:שתיים⁩}
 ```
 
-In the syntax, it's much simpler to include the changes to `name` in that rule,
-rather than patching every place where `name` is used.
-Either way, the parsed value of the name should not include the open/close isolates,
-just as they're not included in the parsed values of quoted literals or quoted patterns.
+Note that the parsed value of the `name` does not include the open/close isolates,
+just as they're not included in the parsed values of quoted literals or quoted patterns,
+even though the production includes the characters.
+We could accomplish this by adding an additional productions to manage `name`, at the cost
+of a more complex ABNF.
+
+**Pros**
+- In the syntax, it's much simpler to include the changes to `name` in the `name` rule,
+  rather than patching every place where `name` is used.
 
+**Cons**
+- Implementations need to remove isolates from the `name` token before comparing
+  the value to other values (such as comparing `function` or `variable` names).
+  Because of namespacing, this requires looking _inside_ the token.
+- Implementations might need to insert isolates when generating names upon serialization.
+  The current data model does not separate `namespace` and `name`,
+  so this might be more complicated.
+- `unquoted-literal` values appear as keys, as operands, and as option values.
+  If not isolated, these can cause spillover effects, so we might need both `name`
+  and `unquoted-literal` isolation.
```

Comment:
Given how `unquoted-literal` is defined as https://github.com/unicode-org/message-format-wg/blob/01f2880bd709653ad850dc746f4a72fe1133c064/spec/message.abnf#L48 and `number-literal` only contains LTR characters https://github.com/unicode-org/message-format-wg/blob/01f2880bd709653ad850dc746f4a72fe1133c064/spec/message.abnf#L50 allowing `name` isolation should cover all the `unquoted-literal` cases that may need isolation.

Reply:
Ur... this is something I don't like, since isolates are "not included" in the `name`, but are baked into the production. I'd rather do more with the ABNF to keep them separate (the isolates and directional characters are in the same places, but part of the surrounding goo instead of being part of the name-like token).

Note that `number-literal` contains only neutral and weakly directional characters (except for `e`, which is strongly LTR). Note that the leading `-` can switch sides in an RTL context (unless isolated or protected with LRM)